Update NFT ref info (#1931)
XRPLF/xrpl-dev-portal: content/references/protocol-reference/transactions/transaction-metadata.md

**Note:** If the modified ledger object has `PreviousTxnID` and `PreviousTxnLgrSeq` fields, the transaction always updates them with the transaction's own identifying hash and the index of the ledger version that included the transaction, but these fields' new value is not listed in the `FinalFields` of the `ModifiedNode` object, and their previous values are listed at the top level of the `ModifiedNode` object rather than in the nested `PreviousFields` object.

## NFT Fields

> An `NFT` transaction response can contain the following fields:

Reviewer: Putting `NFT` in code font makes it seem like you're referring to a specific thing with that exact name, but there isn't one. Maybe soften it up, like so:

```suggestion
Transactions involving NFTs can contain the following fields in the metadata:
```

Also, "can contain" is vague. Is it possible to be more specific about when it does and doesn't?

Based on these fields being lower-case, I am guessing that they are "non-canonical" fields that are calculated at the API layer and are not part of the hashed metadata—is that right? And are they returned by P2P `rippled` servers or only by Clio servers? These details are good to know.

Reviewer 2: These fields are in `rippled@1.11`

> Transactions involving NFTs can contain the following fields in the metadata. These values are added by the Clio server at request time and are not stored in the hashed binary  metadata:

Reviewer 3: extra space  in `binary  metadata`

> Transactions involving NFTs can contain the following fields in the metadata. These values are added by the Clio server at request time and are not stored in the hashed binary  metadata:

Reviewer 3: should we be explicit that these new fields are for responses `tx` and `account_tx` APIs only? People may be confused after they submit a transaction and don't see the field.

Reviewer 2: Will they be present on `nft_history` too?

Reviewer 3: @ckniffen no lol. It's only for rippled. i def forgot about it, it was a while ago. But i think i will create a ticket for it

| Field               | Value                     | Description                |
|:--------------------|:--------------------------|:---------------------------|

> | `nftoken_id`        | String                    | Shows the `NFTokenID` for the `NFToken` that changed on the ledger as a result of the transaction. See [NFTokenID](nftoken.html#nftokenid) |

Reviewer 3: I think we should be more specific here. This metadata field is only present if the tx is a `NFTokenMint` or `NFTokenAcceptOffer`, so that users know that they need to catch this field _only_ for these two transactions.

> | `nftoken_ids`       | Array                     | Shows all the `NFTokenIDs` for the `NFTokens` that changed on the ledger as a result of the transaction. |

Reviewer 3: same reason as above, should specify that this field only applies for `NFTokenCancelOffer`
